src/main/java/Storage.java

```diff
@@ -0,0 +1,139 @@
+import duke.Deadline;
+import duke.Event;
+import duke.Task;
+import duke.Todo;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Scanner;
+import static duke.FileOperations.*;
+
+public class Storage {
+    public static void printLine() {
+        System.out.println("____________________________________________________________");
+    }
+
+    //load the data from the hard disk when Duke starts up
+    //write the existing data into the current list of tasks
+    public static void loadData() throws FileNotFoundException {
+        String filePath = "data/duke.txt";
+        File f = new File(filePath);
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            String str = s.nextLine();
+            convertFromData(str);
+        }
+        System.out.println("Saved data has been loaded!");
+        TaskList.listTasks();
+    }
+
+    public static void load() {
+        try {
+            loadData();
+        } catch (IOException e) {
+            System.out.println("☹ OOPS!!! The file does not exist. Creating a new file.");
+            String pathName = "data";
+            String filePath = "data/duke.txt";
+            File directory = new File(pathName);
+            File file = new File(filePath);
+            if (!directory.exists()) {
+                directory.mkdirs();
+            }
+            try {
+                file.createNewFile();
+                System.out.println("Now your data will be saved in the file. Please proceed!");
+                printLine();
+            } catch (IOException exception) {
+                System.out.println("Error while creating the file.");
+                printLine();
+            }
+        }
+    }
+
+    //save the tasks in the hard disk automatically whenever the task list changes
+    public static void saveData() {
+        String filePath = "data/duke.txt";
+        try {
+            writeToFile(filePath, convertToData(Task.tasks.get(0)) + System.lineSeparator());
+            for (int i = 1; i < Task.tasks.size(); i++) {
+                appendToFile(filePath, convertToData(Task.tasks.get(i)) + System.lineSeparator());
+            }
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    //convert data from text file into Task list format to Load Data
+    public static void convertFromData(String str) {
```

Comment:
I see many magic literals in this function. Maybe use variables in their place? [See here](https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers)